### DISCUSSION TITLE
How to identify the end of sentences in a speech?

### DISCUSSION THREAD

**Discussion by fguidotti-git** [deepgram/community#430] [General help] [answered]
Hi everyone,

I’m having an issue where the audio transcription ends too soon (the endpointing is incorrect).  

I’m streaming my audio from Twilio to Deepgram, and am using a pre-set buffer size to reduce the data size.  I’ve double checked the audio quality, which isn’t an issue.  

What is the best way to identify the end of a sentence in a speech?  My goal is to not pre-set the buffer size, so that it automatically ends when deepgram splits the audio/speech into sentences. 

![image](https://github.com/deepgram/community/assets/38111579/9a15c427-f715-44f8-93c6-8c74b92ee871)

I'm using Deepgram .NET SDK.

**Comment by nikolawhallon** [comment, marked as answer]
There are a few ways to think about this. By default, with `endpointing=true` set (actually by default it's true for streaming anyways), it will send `speech_final` results back as soon as it's detected 10 ms of silence after someone has been speaking. Of course, the speaker could then resume speaking, maybe they were just pausing between words - you can try setting `endpointing=100` or `endpointing=500` to have the algorithm wait for 100 ms or 500 ms (respectively) of silence after a speaker has been speaking to determine that they are done. However, again, they might have just taken a long pause between words. With `punctuate=true` you could look to parse sentences based on the presence of periods, question marks, or exclamation marks too.

At the end of the day, these are all heuristics since a speaker can always resume speaking, creating infinitely long run-on sentences! But hopefully some of these can help with your use-case.
